ui/store/actions.ts

```diff
@@ -3596,17 +3596,14 @@ export function detectTokens(): ThunkAction<
   };
 }
 
-export function detectNfts(): ThunkAction<
-  void,
-  MetaMaskReduxState,
-  unknown,
-  AnyAction
-> {
+export function detectNfts(
+  chainIds: string[],
+): ThunkAction<void, MetaMaskReduxState, unknown, AnyAction> {
```

Comment:
Nice update.

(future) - it seems we always call `detectNfts` with `allChainIds` in multiple files - is this true?
I wonder if this is something we can refactor and reuse this logic.
Either as a hook:
```ts
const useDetectNfts() {
  const dispatch = useDispatch()
  const allChainIds = useSelector(getAllChainsToPoll);
  const callback = useCallback(() => dispatch(detectNfts(allChainIds)), [allChainIds])
  
  return callback;
}
```

or maybe here in the actions thunk?
```ts
export function detectNfts() {
  return async (dispatch, getState) => {
    const chainIds = getAllChainsToPoll(getState());
    await submitRequestToBackground(...)
  }
}
```